### PR TITLE
Install metals automatically 

### LINKED ISSUE #2: Handle Metals installation
@olafurpg Following your proposal [here](https://github.com/scalameta/metals-sublime/pull/1#issuecomment-531574101)

Shall we assume that coursier is installed and available in the classpath ? like in the vscode plugin?
Or should we embed some kind of coursier  luncher like here https://github.com/tomv564/LSP-dotty

How are the server updates handled in the vscode plugin ? do you have to change the version by hand every time there is a server release ?

Just want to understand and document what is the expected workflow 

cc @gabro

### PR DESCRIPTION
**Pull request by ayoub-benali**
closes #2 

Mostly inspired by https://github.com/scalameta/metals-vscode/blob/f6b8b193ea008da74af1b78d5bc8594393a13752/src/extension.ts#L103

One difference is that we use `coursier launch` command directly to fetch Metals artifact.  

- [x] provide a way to specify java home, server version and server properties
- [x] embed coursier
- [x] make metals actually start
- [x] handle fetching options
- [x] handle java options

### CODE REVIEW

File: plugin.py [hunk 1]
```diff
@@ -1,14 +1,61 @@
...
+    if not java_path :
+        window.status_message("Please install java or set the 'java_home' setting")
+
+    server_version = plugin_settings.get('server_version')
+    if not server_version :
+        window.status_message("'server_version' should be set")
+
+    server_properties = prepare_server_properties(plugin_settings.get('server_properties', []))
+
+    service_path = [
```
Comment (ayoub-benali) [inline plugin.py]: @tomv564 
I need to access the plugin settings in order to construct the  `binary_args` array for the client.
which means there could be a case where the client would have an empty array :/ do you know a better way to do this ?

**Comment by ayoub-benali** [comment]
@olafurpg I tried to mimic the way you do things in the vs code extension and managed to create the coursier command handling some settings already. But the server isn't starting even though I see the artifact downloaded the first time in the logs.

`coursier launch` didn't help either. Could I be missing something ?

File: plugin.py [hunk 2]
```diff
@@ -1,14 +1,61 @@
...
+        "sonatype:public",
+        "-r",
+        "sonatype:snapshots",
+        "-p"
+    ]
+
 metals_config = ClientConfig(
     name=config_name,
-    binary_args=[service_name],
+    binary_args=service_path,
```
Comment (olafurpg) [inline plugin.py]: The VS Code extension launches `java` twice:
```
metals_classpath=$(java -jar coursier fetch org.scalameta:metals...)
java -classpath $metals_classpath scala.meta.metals.Main
```
The command that Sublime LSP should invoke is the second command with the Metals Main.

**Comment by ayoub-benali** [comment]
@MasseGuillaume Hi, do you mind giving this a try ?

**Comment by ayoub-benali** [comment]
@olafurpg we just have to make a PR to package control repo: https://packagecontrol.io/docs/submitting_a_package#Step_6

**Pull request merged**